src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,141 @@
+package duke;
+
+class Parser {
+
+    private String desc;
+    private Task task = null;
+    private static int count = 0;
+
+    /**
+     * Constructor for a Parser object used to process input and strings.
+     */
+    Parser(){};
+
+    /**
+     * Takes in line of information to process into commands for program to read.
+     * @param line task description or information needed for processing.
+     * @return command used for updating tasks.
+     * @throws InvalidCommandException when program gives an invalid command type.
+     */
+    Command process(String line) throws InvalidCommandException {
+        String[] commands = line.split(" ");
+        String first = commands[0];
+        try {
+            switch (first) {
+            case "bye":
+                return new Command(CommandType.EXIT);
+            case "list":
+                return new Command(CommandType.PRINTLIST);
+            case "todo":
+            case "deadline":
+            case "event":
+                return new Command(CommandType.ADD, line);
+            case "done":
+                return new Command(CommandType.DONE, line);
+            case "delete":
+                return new Command(CommandType.DELETE, line);
+            case "find":
+                return new Command(CommandType.FIND, line);
+            default:
+                throw new InvalidCommandException();
+            }
+        } catch (DukeException e) {
+            e.printError();
+        }
+        return new Command();
```

Comment:
This line may not be executed even after u catch the error. (see [Flow control in try catch finally in Java](https://www.geeksforgeeks.org/flow-control-in-try-catch-finally-in-java/)). You may want to consider using the finally block for this line (as below)

`finally {
return new Command();
}`